### PR TITLE
feat(contracts): refactor contracts wrappers

### PR DESCRIPTION
**Pull request by nduchak**

### CODE REVIEW

File: src/contracts/tokens/rif/erc20.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import { TransactionReceipt } from 'web3-eth'
+import { AbiItem } from 'web3-utils'
+
+import ERC20 from '../../abi/ERC20.json'
+import { rifTokenAddress } from '../../config'
+import ContractBase from '../../wrappers/contract-base'
+import {
+  ERC20ContractI, TransactionOptions,
+} from '../../interfaces'
+
```
Comment (jurajpiar) [inline src/contracts/tokens/rif/erc20.ts:11]: absolute paths

File: src/contracts/tokens/rif/erc677.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import ERC677 from '@rsksmart/erc677/ERC677Data.json'
+import Web3 from 'web3'
+import { AbiItem } from 'web3-utils'
+import { TransactionReceipt } from 'web3-eth'
+
+import { rifTokenAddress } from '../../config'
+import ContractBase from '../../wrappers/contract-base'
+import { TransactionOptions } from '../../interfaces'
```
Comment (jurajpiar) [inline src/contracts/tokens/rif/erc677.ts:8]: absolute paths

File: src/contracts/wrappers/contract-base.ts [hunk 3]
```diff
@@ -0,0 +1,90 @@
+import Web3 from 'web3'
+import { TransactionReceipt } from 'web3-eth'
+import { Contract } from 'web3-eth-contract'
+
+import Logger from '../../utils/Logger'
+import { TxOptions } from '../interfaces'
+import withWaitForReceipt from '../utils'
```
Comment (jurajpiar) [inline src/contracts/wrappers/contract-base.ts:7]: absolute paths

**Review by jurajpiar — CHANGES_REQUESTED** [review]
Looks good.
It's only so many comments because of absolute paths and because I commented on older issues that, however are included in this PR.

File: src/contracts/rns/Marketplace.ts [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+      { ...txOptions, gasMultiplier: 1.1 },
+    )
+  }
+
+  // Unplace: Proxy function for unplacement transaction
+  public unplace(
+    tokenId: string,
+    txOptions: TransactionOptions,
+  ): Promise<TransactionReceipt> {
+    const unplaceTx = this.contract.methods.unplace(tokenId)
```
[the remark this answers is not in the document]
Comment (nduchak) [reply src/contracts/rns/Marketplace.ts:49]: Good idea

File: src/contracts/storage/Staking.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+import { AbiItem } from 'web3-utils'
+import { stakingAddress, StorageSupportedTokens } from '../config'
+import { SUPPORTED_TOKENS, TxOptions } from '../interfaces'
+import { getTokens } from '../utils'
+import ContractWithTokens from '../wrappers/contract-using-tokens'
+
+export type StorageStakingContractErrorId = 'contract-storage-staking'
+
+const zeroBytes = '0x'.padEnd(64, '0')
+const extraGasPercentage = 1.3
```
[the remark this answers is not in the document]
Comment (nduchak) [reply src/contracts/storage/Staking.ts]: I just move it as a static property of Contract

File: src/ui-config.json [hunk 6]
```diff
@@ -4,14 +4,18 @@
             "rif": "0x67b5656d60a809915323bf2c40a8bef15a152e3e",
             "rnsDotRskOwner": "0x4bf749ec68270027C5910220CEAB30Cc284c7BA2",
             "marketplace": "0x12ae8689c3E889A0fC81f4E2ba9D6C5Bf358F392",
-            "storageManager": "0xeea2Fc1D255Fd28aA15c6c2324Ad40B03267f9c5",
-            "storageStaking": "0x7231ECd1355a60251eE56Bf81f987969fc9bAe29"
+            "storageManager": "0x8b3013089671aDDc15744D2F2725E7ea3D334939",
+            "storageStaking": "0x99750Da185B646E56DdBDc80dd3263e2166bcb3f"
```
[the remark this answers is not in the document]
Comment (nduchak) [reply src/ui-config.json]: Reverted

File: src/contracts/storage/Staking.ts [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+import { TransactionReceipt } from 'web3-eth'
+import { AbiItem } from 'web3-utils'
+import { stakingAddress, StorageSupportedTokens } from '../config'
+import { SUPPORTED_TOKENS, TxOptions } from '../interfaces'
+import { getTokens } from '../utils'
+import ContractWithTokens from '../wrappers/contract-using-tokens'
+
+export type StorageStakingContractErrorId = 'contract-storage-staking'
+
+const zeroBytes = '0x'.padEnd(64, '0')
```
[the remark this answers is not in the document]
Comment (nduchak) [reply src/contracts/storage/Staking.ts:14]: Done

**Review by jurajpiar — APPROVED** [review]
👍

**Review by farinalvaro — APPROVED** [review]
LRGTM - Nothing to add after Juraj's review 🕵️‍♂️

**Pull request merged**